### PR TITLE
Use Color#mix for color interpolation

### PR DESCRIPTION
**Pull request by mitioshi**
### Description
This PR changes the algorithm used for color interpolation to use `i_slint_core::graphics::color::Color#mix`. 
This would resolve the issue of color transitions having incorrect values when animating color to/from `transparent`

## Changes
Color#mix accounts for the alpha channel, so a transparent transition is not skewed by changes in transparency.
For example, with the proposed changes, a transition from `transparent` to `lightgray` changes like this .
On the left is the current algorithm, on the right is Color#mix
image

### Motivation
Motivated by https://github.com/slint-ui/slint/issues/5055

### CODE REVIEW

**Comment by ogoffart** [comment]
Thanks a lot for the PR.

Looks good.  Did you check that this is the right direction and the animation looks nice with this patch?  (and it shouldn't need to be `1-t`?)

(Don't worry about the node test failling on Linux on the CI, this is a known flaky test)

File: internal/core/graphics/color.rs [hunk 1]
```diff
@@ -305,12 +305,7 @@ impl Color {
 
 impl InterpolatedPropertyValue for Color {
     fn interpolate(&self, target_value: &Self, t: f32) -> Self {
-        Self {
-            red: self.red.interpolate(&target_value.red, t),
-            green: self.green.interpolate(&target_value.green, t),
-            blue: self.blue.interpolate(&target_value.blue, t),
-            alpha: self.alpha.interpolate(&target_value.alpha, t),
-        }
+        self.mix(target_value, t)
```
Comment (ogoffart) [inline internal/core/graphics/color.rs:308]: Actually, I checked, and it seems like this should be reversed:

```suggestion
        target_value.mix(self, t)
```

(The mix argument order is a bit misleading)

Comment (mitioshi) [reply internal/core/graphics/color.rs]: Fixed this. The test I added initially didn't catch it since it only checked the middle point for mixing =)

**Review by ogoffart — APPROVED** [review]
great, thank you!

**Pull request merged**